### PR TITLE
Summary updates and adding simple pages for quick test

### PR DESCRIPTION
**Pull request by ivantcholakov**
* I removed a wrong idea of mine within REST_Controller and updated the example API accordingly. URIs like users/{id} as an alternative of the ordinary users/id/{id} should be supported by using routing rules. I added a configuration routes.php for such a demonstration.
* I suppressed some possible "array to string conversion" notices within Format.php.
* I added a modified "Welcome" page and a page for quick testing the REST server (a list of sample links). This is for convenience, otherwise people are not motivated to test.

### CODE REVIEW

File: application/controllers/api/Key.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ public function level_post()
     }
 
     /**
-     * Change the level
+     * Suspent a key
```
Comment (softwarespot) [inline application/controllers/api/Key.php]: suspent should be suspend

File: application/views/rest_server.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        $("#ajax").on("click", function(evt) {
+
+            evt.preventDefault();
+
+            $.ajax({
+
+                url: $(this).attr("href"), // URL from the link that was clicked on.
+
+                success: function(data, textStatus, jqXHR) {
```
Comment (softwarespot) [inline application/views/rest_server.php]: Deprecation Notice: The jqXHR.success(), jqXHR.error(), and jqXHR.complete() callbacks are deprecated as of jQuery 1.8. To prepare your code for their eventual removal, use jqXHR.done(), jqXHR.fail(), and jqXHR.always() instead.

Source: http://api.jquery.com/jquery.ajax/

File: application/views/rest_server.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+                    // The 'data' parameter is an array of objects that can be looped over.
+
+                    if (window.JSON && window.JSON.stringify) {
```
Comment (softwarespot) [inline application/views/rest_server.php]: This of course is my personal preference, but I don't normally do this for JSON.stringify(), as according to [caniuse.com](http://caniuse.com/#search=JSON), it's more than likely most users will have a version that includes this.

File: application/controllers/api/Key.php [hunk 4]
```diff
@@ -233,16 +233,16 @@ private function _generate_key()
...
+            ->row();
     }
 
     private function _key_exists($key)
     {
         return $this->db
-			->where(config_item('rest_key_column'), $key)
-			->count_all_results(config_item('rest_keys_table')) > 0;
+            ->where(config_item('rest_key_column'), $key)
+            ->count_all_results(config_item('rest_keys_table')) > 0;
```
Comment (softwarespot) [inline application/controllers/api/Key.php]: What IDE do you use? It's not a problem, just curious. I use a mixture of Netbeans PHP and SublimeText for web development.

**Comment by ivantcholakov** [comment]
Netbeans and Meld for comparing sources.

File: application/controllers/api/Key.php (hunk 1 above)
Comment (ivantcholakov) [inline application/controllers/api/Key.php]: https://github.com/ivantcholakov/codeigniter-restserver/commit/9d87d1355da6eef7c546506c429035153b815ade

**Comment by softwarespot** [comment]
Thanks. A lot of work there. Thank you.

File: application/views/rest_server.php (hunk 2 above)
Comment (ivantcholakov) [inline application/views/rest_server.php]: https://github.com/ivantcholakov/codeigniter-restserver/commit/1c8cd8fa93d93d3cfe97a7042e960e24f6973bbf

File: application/views/rest_server.php (hunk 3 above)
Comment (ivantcholakov) [inline application/views/rest_server.php]: Maybe I am getting old. :-) I will remove this check for the sake of simplicity.

Comment (ivantcholakov) [inline application/views/rest_server.php]: https://github.com/ivantcholakov/codeigniter-restserver/commit/de4d8d17884998b302e3f95c4b3fd87c55b33c33

**Pull request merged**